vcr/request.py

```diff
@@ -32,9 +33,19 @@ def __init__(self, method, uri, body, headers):
             self.body = body.read()
         else:
             self.body = body
-        self.headers = {}
-        for key in headers:
-            self.add_header(key, headers[key])
+        self.headers = CaseInsensitiveDict()
```

Comment:
or better yet, actually: doesn't your setter actually make it so you don't need a setter here at all?

i.e. just leave `self.headers = headers` and everything should just work